### DISCUSSION TITLE
replace_string question

### DISCUSSION THREAD

**Discussion by yosiasz** [grafana/grafana-infinity-datasource#332] [General]
Hello there,

Using both 0.8.8 and 1.0.0 plugin and following sample data

```
[
  {
    "userID": 58,
    "Start": "2022-06-19T03:44:20.220Z",
    "End": "2022-06-19T13:44:20.220Z"
  },
  {
    "userID": 62,
    "Start": "2022-06-20T04:44:20.220Z",
    "End": "2022-06-20T14:44:20.220Z"
  },
  {
    "userID": 78,
    "Start": "2022-06-20T02:04:20.220Z",
    "End": "2022-06-20T17:04:20.220Z"
  }
]
```

UQL

```
parse-json
| extend "zaza"=replace_string("userID",58, 'Zoro')
```

It seems like it is not working . Even if I convert the userID to string.

Any idea why?

Thanks

**Comment by yosiasz** [comment]
it ignores grafana's convert to String but this works. Yaay!

```
parse-json
| extend "userID"=tostring("userID") 
| extend "userID"=replace_string("userID",'58','ዮስያስ')
```

**Reply by yosiasz** [reply]
this is hands down THE BEST plugin on this planet. Great work!!

**Reply by yesoreyeram** [reply]
What do you mean by it ignore grafana covert to string?

**Reply by yosiasz** [reply]
I used grafana's own convert transformation and it seems like infinity does not see it as string. But using the tostring function works. Odd
